test/logstream/kubelogs.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logstream
+
+import (
+	"bufio"
+	"encoding/json"
+	"log"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/knative/pkg/ptr"
+	"github.com/knative/pkg/test"
+	servingtest "github.com/knative/serving/test"
+)
+
+type kubelogs struct {
+	Namespace string
+
+	once sync.Once
+	m    sync.RWMutex
+	keys map[string]logger
+}
+
+type logger func(string, ...interface{})
+
+var _ streamer = (*kubelogs)(nil)
+
+// Init implements streamer
+func (k *kubelogs) Init() {
+	kc, err := test.NewKubeClient(test.Flags.Kubeconfig, test.Flags.Cluster)
+	if err != nil {
+		log.Fatalf("error loading client config: %v", err)
```

Comment:
```suggestion
		log.Fatalf("Error loading client config: %v", err)
```

Reply:
I presume all these fatals came from kubectl, but I doubt we really want to `log.Fatal`, just because we failed to stream the logs...

Reply:
Agreeing on the `Fatalf` usage.

Reply:
IDK which of us @markusthoemmes is agreeing with.  IDK that I feel strongly here, so I'm happy to relax things, but wanted to check first.

Reply:
Actually, now that it's a `sync.Once` we can pass in `*testing.T` and use `t.Errorf()`, I'll do that.  The failing test will vary, but it won't kill the whole test process.  We can see if this is too noisy.